APIs/openEO/authentication.qmd

```diff
@@ -0,0 +1,226 @@
+[`Connection`](https://open-eo.github.io/openeo-python-client/api.html#openeo.rest.connection.Connection){target="_blank"}.
+
+:::
+
+
+
+### Client Credentials Flow
+
+
+OpenID Connect also supports the so-called "client credentials flow",
+which is a non-interactive flow, based on a client id and a secret,
+that is suitable for machine-to-machine authentication.
+
+The openEO Python client library has built-in support for this flow,
+with the ['authenticate_oidc_client_credentials()` method](https://open-eo.github.io/openeo-python-client/auth.html#oidc-authentication-client-credentials-flow){target="_blank"}:
+
+```python
+connection.authenticate_oidc_client_credentials(
+    client_id=...,
+    client_secret=...,  # Note: Do not hard-code your secret here!
+)
+```
+
+#### Obtaining Client Credentials
+
+::: {.callout-warning}
+Usage of client credentials in openEO is an experimental feature,
+it's not widely supported across openEO backends,
+and the setup procedure is not fully standardized or streamlined yet.
+:::
+
+The Sentinel Hub service in the Copernicus Data Space Ecosystem
+has a [dashboard web app](https://shapps.dataspace.copernicus.eu/dashboard)
+and under the account settings there is a self-service feature to
+[register your own OAuth client](../SentinelHub/Overview/Authentication.qmd#registering-oauth-client){target="_blank"}.
+The client id and client secret you obtain here can also be used
+for the client credentials flow with the openEO service of Copernicus Data Space Ecosystem.
+
+#### Caveats And Considerations
+
+-   Treat your client secret securely, like a password.
+    Take extra care to not leak it accidentally.
+    For example, the simplicity of the `authenticate_oidc_client_credentials()` example snippet above,
+    make it very tempting to just hard-code the client secret in your scripts or notebooks,
+    which might then eventually be stored for eternity in some version control repository.
+
+    Instead, read the client secret from a secure location
+    (e.g. a private file outside the reach of your version control repositories),
+    or leverage environment variables (e.g as directly [supported by the openEO Python client library](https://open-eo.github.io/openeo-python-client/auth.html#oidc-client-credentials-using-environment-variables){target="_blank"}).
+-   The client credentials only identify your (OAuth) client, not your personal user account.
+
+    -   This means that openEO resources such as openEO batch jobs, their results, UDP's, etc
+        from one identity are not available to the other.
+        For example, you can not list the batch jobs created with client credentials,
+        when you are using your personal account.
+    -   Likewise, the balances of processing credits are separate.
+        It is however possible to link the balance of the client credentials account
+        to your personal account.
+        To enable that: contact support and provide your client id and user id.
```

Comment:
@JanssenBrm  @rahmandawibowo-vito  this probably needs some more work. I'm not sure what you need nowadays at the marketplace side of things to link client credential accounts with user accounts